x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/security/authc/AuthenticationTests.java

```diff
@@ -923,6 +926,31 @@ public void testMaybeRewriteMetadataForCrossClusterAccessAuthentication() throws
         assertThat(innerRewritten, equalTo(innerAuthentication.maybeRewriteForOlderVersion(olderVersion)));
     }
 
+    public void testCopyWithFilteredMetadataFields() {
+        final Map<String, Object> metadataToKeep = ESTestCase.randomMap(
+            0,
+            10,
+            () -> new Tuple<>(ESTestCase.randomAlphaOfLength(20), ESTestCase.randomAlphaOfLength(10))
+        );
+        final Map<String, Object> allMetadata = new HashMap<>(metadataToKeep);
+        allMetadata.putAll(
+            ESTestCase.randomMap(0, 10, () -> new Tuple<>(ESTestCase.randomAlphaOfLength(30), ESTestCase.randomAlphaOfLength(10)))
+        );
+        // Using service account since it allows us to easily specific metadata, but will not fail instantiation if any service-account
+        // specific metadata is missing
+        final Authentication authentication = AuthenticationTestHelper.builder().serviceAccount().metadata(Map.copyOf(allMetadata)).build();
```

Comment:
Not like it is important, but the fact we don't check service-account specific metadata could be a separate bug. No need to do anything here.

Reply:
Ya agreed, we probably want to enforce this as a consistency check but I didn't look too closely.